build_daemon/lib/client.dart

```diff
@@ -43,10 +43,35 @@ Future<void> _handleDaemonStartup(
 
   // The daemon may log critical information prior to it successfully
   // starting. Capture this data and forward to the logHandler.
+  //
+  // Whenever we see a `logStartMarker` we will parse everything between that
+  // and the `logEndMarker` as a `ServerLog`. Everything else is considered a
+  // normal INFO level log.
+  StringBuffer nextLogRecord;
```

Comment:
Would it be sensible to write a `StreamTransformer<String, String>` that collects between the start and end markers so that we separate out that logic from how we are handling log records?

Reply:
We could do that, it would also allow us to unit test it...